src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/DeprecationCapability.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using NuGet.Versioning;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class DeprecationCapability : IDeprecated
+    {
+        private static readonly string StarAll = VersionRangeFormatter.Instance.Format("p", VersionRange.Parse("*"), VersionRangeFormatter.Instance);
+        private static readonly string StarAllFloating = VersionRangeFormatter.Instance.Format("p", VersionRange.Parse("*-*"), VersionRangeFormatter.Instance);
+
+        public DeprecationCapability(PackageDeprecationMetadataContextInfo deprecatedInfo)
+        {
+            _deprecated = deprecatedInfo;
+        }
+
+        private PackageDeprecationMetadataContextInfo _deprecated;
+        public PackageDeprecationMetadataContextInfo DeprecationMetadata
+        {
+            get => _deprecated;
+            private set => _deprecated = value;
+        }
+
+        public bool IsDeprecated => DeprecationMetadata != null;
+        public string? AlternatePackageText
+        {
+            get
+            {
+                if (DeprecationMetadata.AlternatePackage == null)
+                {
+                    return null;
+                }
+
+                // pretty print
+                string versionString = VersionRangeFormatter.Instance.Format("p", DeprecationMetadata.AlternatePackage.VersionRange, VersionRangeFormatter.Instance);
+
+                if (StarAll.Equals(versionString, StringComparison.InvariantCultureIgnoreCase) || StarAllFloating.Equals(versionString, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return DeprecationMetadata.AlternatePackage.PackageId;
+                }
+
+                return $"{DeprecationMetadata.AlternatePackage.PackageId} {versionString}";
```

Comment:
The model should not be responsible for presentation, only representation of state. Pretty printing the alternate package is the responsibility of the View with help from the ViewModel, if necessary. Perhaps we want to expose whether the version string matches the StarAll pattern or the StarAllFloating, and the Alternate package. That part can be considered business logic. How it then gets displayed and concatenated for display purposes is a presentation concern best handled in the View with possible help from the ViewModel.